driver/sources/src/main/scala/org/virtuslab/ideprobe/WorkspaceProvider.scala

```diff
@@ -58,8 +58,11 @@ object WorkspaceProvider {
 
 sealed trait WorkspaceTemplate extends WorkspaceProvider {
 
-  override final def setup(): Path = {
-    val workspaceBase = Files.createTempDirectory("ideprobe-workspace")
+  override final def setup(paths: PathsConfig): Path = {
+    val dirName = "ideprobe-workspace"
+    val suffix = scala.util.Random.nextLong().abs.toString
+    val workspaceBase = paths.workspaces.resolve(dirName + suffix)
```

Comment:
It would be best to add to PathExtensions method like createTempDirectory(prefix: String = "", suffix: String = ""). I believe it is safer to use UUID.randomUUID() rather than Random.nextLong()